File: test/AirVisual-test.js
// Run from command line via
// $ mocha AirVisual-test.js --timeout=20000
// see AirVisualTest.java for Rosetta stone

const webdriver = require('selenium-webdriver');

const By = webdriver.By;       // DRY
const until = webdriver.until;

const driver = new webdriver.Builder()
    .forBrowser('chrome')
    .build();

describe('AirVisual Widget', () => {

  process.on('unhandledRejection', error => {
    throw(error); // promote promise warning to mocha error
  });
  
  beforeEach(async function() {
    await driver.navigate().to('https://www.sweetlightstudios.com/contact');
    await driver.wait(until.titleContains('South San Francisco'),10000);
  });

  it('Uses widget to nav to AirVisual', async function() {
    await driver.findElement(By.css('article.contact'));
    const widget = await driver.findElement(By.name('airvisual_widget'));
    await driver.wait(until.elementIsVisible(widget),10000);
    await driver.wait(until.elementIsEnabled(widget),10000);
    await widget.click();
    await driver.wait(until.titleContains('AirVisual - Real-time air quality'),10000);
    await driver.navigate().back();
    await driver.wait(until.titleContains('South San Francisco'),10000);
    let element = await driver.findElement(By.css("article.contact"));
    await driver.wait(until.elementIsVisible(element),10000)
    }); 

  afterEach(async function() {
    await driver.quit();
  });

});

File: test/AirVisualTest.java
import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import static junit.framework.TestCase.assertEquals;
import static org.junit.Assert.assertTrue;

public class AirVisualTest {
    static WebDriver driver;

    @BeforeClass
    static public void beforeClass() {
        driver = new ChromeDriver();
        driver.get("https://www.sweetlightstudios.com/contact/");
        String titleText = driver.getTitle();
        System.out.println("titleText: " + titleText);
    }

    @Test
    public void airvisualWidgetButtonTest(){
        driver.findElement(By.cssSelector("article.contact")).isDisplayed(); // contact article
        WebElement widget = driver.findElement(By.name("airvisual_widget"));
        new WebDriverWait(driver,10).until(ExpectedConditions.elementToBeClickable(widget));
        widget.click();
        new WebDriverWait(driver,10).until(ExpectedConditions.titleContains("AirVisual - Real-time air quality"));
        driver.navigate().back(); // come back to Sweet Light Studios
        new WebDriverWait(driver,10).until(ExpectedConditions.titleContains("South San Francisco"));
        driver.findElement(By.cssSelector("article.contact")).isDisplayed(); // verify contact article is displayed
    }

    @AfterClass
    static public void afterClass() {
        driver.quit();
    }
}
